### ISSUE TITLE
support webvtt subs in mkv

### ISSUE THREAD

**Issue by latot** [mpv-player/mpv#3247]
Hi, well basically mpv don't support show webvtt subtitles merged into a mkv file, externally the subs works fine.

```
[mkv] Subtitle type 'S_TEXT/WEBVTT' is not supported.
 (+) Video --vid=1 (*) (h264)
 (+) Audio --aid=1 (*) (aac)
 (+) Subs  --sid=1 (*) ()
[sub/ass] Could not open libavcodec subtitle converter
[sub] Could not find subtitle decoder for format ''.
```

with -v

```
[cplayer] Command line options: '-v' 'some.mkv'
[cplayer] mpv git-3682df2 (C) 2000-2016 mpv/MPlayer/mplayer2 projects
[cplayer]  built on UNKNOWN
[cplayer] ffmpeg library versions:
[cplayer]    libavutil       54.31.100
[cplayer]    libavcodec      56.60.100
[cplayer]    libavformat     56.40.101
[cplayer]    libswscale      3.1.101
[cplayer]    libavfilter     5.40.101
[cplayer]    libswresample   1.2.101
[cplayer] ffmpeg version: 2.8.6
[cplayer] 
[cplayer] Configuration: /var/tmp/portage/media-video/mpv-9999/work/mpv-9999/waf --prefix=/usr --libdir=/usr/lib64 --confdir=/etc/mpv --docdir=/usr/share/doc/mpv-9999 --enable-libmpv-shared --disable-libmpv-static --disable-build-date --disable-optimize --disable-debug-build --disable-test --disable-pdf-build --disable-vf-dlopen-filters --enable-zsh-comp --enable-iconv --enable-libguess --enable-libsmbclient --enable-lua --enable-libass --enable-libass-osd --enable-encoding --enable-libbluray --enable-dvdread --enable-dvdnav --enable-cdda --enable-enca --disable-rubberband --enable-lcms2 --disable-vapoursynth --disable-vapoursynth-lazy --enable-libavdevice --lua=luajit --enable-uchardet --enable-sdl2 --disable-sdl1 --disable-oss-audio --disable-rsound --enable-pulse --enable-jack --enable-openal --enable-alsa --enable-wayland --enable-x11 --enable-xss --enable-xext --enable-xv --enable-xinerama --enable-xrandr --enable-gl-x11 --disable-egl-x11 --enable-gl-wayland --enable-gl --disable-vdpau --disable-vdpau-gl-x11 --disable-vaapi --disable-vaapi-glx --disable-caca --disable-drm --enable-jpeg --disable-rpi --disable-vaapi-hwaccel --disable-tv --disable-tv-v4l2 --disable-libv4l2 --enable-dvbin configure
[cplayer] List of enabled features: alsa any-gl asm atomics av-pix-fmt-mmal av-version-info avcodec-chroma-pos-api avframe-metadata avframe-skip-samples c11-tls cdda cplayer dlopen dvbin dvdnav dvdread enca encoding fchmod gbm gbm.h gl gl-wayland gl-x11 glibc-thread-name glob iconv jack jpeg lcms2 libass libass-osd libav libavdevice libavfilter libbluray libdl libguess libm libmpv-shared librt libsmbclient libswresample linux-fstatfs lua nanosleep openal plain-gl posix posix-or-mingw posix-spawn pthreads pulse resampler sdl2 shm standard-gl stdatomic subprocess termios uchardet videodev vt.h wayland x11 xext xinerama xrandr xss xv zlib zsh-comp
[global] config path: '' -> '/home/pipe/.config/mpv'
[global] config path: 'mpv.conf' -/-> '/home/pipe/.config/mpv/mpv.conf'
[global] config path: 'config' -/-> '/home/pipe/.config/mpv/config'
[global] config path: 'mpv.conf' -/-> '/home/pipe/.mpv/mpv.conf'
[global] config path: 'config' -/-> '/home/pipe/.mpv/config'
[global] config path: 'mpv.conf' -/-> '/etc/mpv/mpv.conf'
[global] config path: 'config' -/-> '/etc/mpv/config'
[cplayer] Setting option 'v' = '' (flags = 8)
[global] config path: 'input.conf' -/-> '/home/pipe/.config/mpv/input.conf'
[global] config path: 'input.conf' -/-> '/home/pipe/.mpv/input.conf'
[global] config path: 'input.conf' -/-> '/etc/mpv/input.conf'
[input] Falling back on default (hardcoded) input config
[osc] Loading script @osc.lua...
[global] config path: 'scripts' -/-> '/home/pipe/.config/mpv/scripts'
[global] config path: 'scripts' -/-> '/home/pipe/.mpv/scripts'
[global] config path: 'scripts' -/-> '/etc/mpv/scripts'
[osc] loading mp.defaults
[osc] loading @osc.lua
[global] config path: 'lua-settings/osc.conf' -/-> '/home/pipe/.config/mpv/lua-settings/osc.conf'
[global] config path: 'lua-settings/osc.conf' -> '/home/pipe/.mpv/lua-settings/osc.conf'
[cplayer] Run command: define-section, flags=0, args=[showhide, mouse_move script-binding osc/__keybinding1
[cplayer] mouse_leave script-binding osc/__keybinding2
[cplayer] , force]
[cplayer] Run command: enable-section, flags=0, args=[showhide, allow-hide-cursor+allow-vo-dragging]
[cplayer] Run command: define-section, flags=0, args=[input, mouse_btn0 script-binding osc/__keybinding3
[cplayer] shift+mouse_btn0 script-binding osc/__keybinding4
[cplayer] mouse_btn2 script-binding osc/__keybinding5
[cplayer] mouse_btn0_dbl ignore
[cplayer] shift+mouse_btn0_dbl ignore
[cplayer] mouse_btn2_dbl ignore
[cplayer] , force]
[cplayer] Run command: enable-section, flags=0, args=[input, ]
[cplayer] Run command: define-section, flags=0, args=[input_osc, del script-binding osc/__keybinding6
[cplayer] , default]
[cplayer] Run command: enable-section, flags=0, args=[input_osc, allow-hide-cursor+allow-vo-dragging]
[cplayer] Run command: define-section, flags=0, args=[input_forced_osc, , force]
[cplayer] Run command: enable-section, flags=0, args=[input_forced_osc, allow-hide-cursor+allow-vo-dragging]
[cplayer] Done loading @osc.lua.
[ytdl_hook] Loading script @ytdl_hook.lua...
[global] config path: 'scripts' -/-> '/home/pipe/.config/mpv/scripts'
[global] config path: 'scripts' -/-> '/home/pipe/.mpv/scripts'
[global] config path: 'scripts' -/-> '/etc/mpv/scripts'
[ytdl_hook] loading mp.defaults
[cplayer] Run command: disable-section, flags=0, args=[input]
[global] config path: 'fonts' -/-> '/home/pipe/.config/mpv/fonts'
[global] config path: 'fonts' -/-> '/home/pipe/.mpv/fonts'
[global] config path: 'fonts' -/-> '/etc/mpv/fonts'
[osd/libass] Shaper: FriBidi 0.19.6 (SIMPLE) HarfBuzz-ng 1.2.7 (COMPLEX)
[global] config path: 'subfont.ttf' -/-> '/home/pipe/.config/mpv/subfont.ttf'
[global] config path: 'subfont.ttf' -/-> '/home/pipe/.mpv/subfont.ttf'
[global] config path: 'subfont.ttf' -/-> '/etc/mpv/subfont.ttf'
[global] config path: 'fonts.conf' -/-> '/home/pipe/.config/mpv/fonts.conf'
[global] config path: 'fonts.conf' -/-> '/home/pipe/.mpv/fonts.conf'
[global] config path: 'fonts.conf' -/-> '/etc/mpv/fonts.conf'
[osd/libass] Setting up fonts...
[ytdl_hook] loading @ytdl_hook.lua
[cplayer] Run command: hook-add, flags=0, args=[on_load, 1, 10]
[cplayer] Done loading @ytdl_hook.lua.
[global] config path: 'scripts' -/-> '/home/pipe/.config/mpv/scripts'
[global] config path: 'scripts' -/-> '/home/pipe/.mpv/scripts'
[global] config path: 'scripts' -/-> '/etc/mpv/scripts'
[global] config path: 'watch_later' -> '/home/pipe/.config/mpv/watch_later'
[cplayer] Playing: re_hamatora.mkv
[cplayer] Running hook: ytdl_hook/on_load
[osd/libass] Using font provider fontconfig
[osd/libass] Done.
[cplayer] Run command: hook-ack, flags=0, args=[on_load]
[ifo] Opening re_hamatora.mkv
[ifo/dvdnav] Opening re_hamatora.mkv
[bdmv/bluray] Opening re_hamatora.mkv
[file] Opening re_hamatora.mkv
[file] Stream opened successfully.
[demux] Trying demuxers for level=normal.
[mkv] Found the head...
[mkv] + a segment...
[mkv] Parsing seek head...
[mkv] |+ segment information...
[mkv] | + muxing app: libebml v1.3.3 + libmatroska v1.4.4
[mkv] | + writing app: mkvmerge v9.2.0 ('Photograph') 64bit
[mkv] | + timecode scale: 1000000
[mkv] | + duration: 89.834s
[mkv] | + segment uid c6 b2 1a 0a de 02 58 76 dd a1 bd 96 31 e0 d3 72
[mkv] |+ segment tracks...
[mkv] | + a track...
[mkv] |  + Track number: 1
[mkv] |  + Track type: Video
[mkv] |  + Video track
[mkv] |   + Display width: 1280
[mkv] |   + Display height: 720
[mkv] |   + Pixel width: 1280
[mkv] |   + Pixel height: 720
[mkv] |  + Codec ID: V_MPEG4/ISO/AVC
[mkv] |  + CodecPrivate, length 48
[mkv] |  + Language: und
[mkv] |  + Default duration: 41.667ms ( = 24.000 fps)
[mkv] | + a track...
[mkv] |  + Track number: 2
[mkv] |  + Track type: Audio
[mkv] |  + Audio track
[mkv] |   + Sampling frequency: 48000.000000
[mkv] |   + Channels: 2
[mkv] |  + Codec ID: A_AAC
[mkv] |  + CodecPrivate, length 2
[mkv] |  + Language: und
[mkv] |  + Default duration: 21.333ms ( = 46.875 fps)
[mkv] | + a track...
[mkv] |  + Track number: 3
[mkv] |  + Track type: Subtitle
[mkv] |  + Codec ID: S_TEXT/WEBVTT
[mkv] |  + CodecPrivate, length 34
[mkv] |  + Language: und
[mkv] |+ found cluster
[mkv] Deferring reading cues.
[mkv] Seeking to 29487895 to read header element 0x1254c367.
[mkv] All headers are parsed!
[mkv] Subtitle type 'S_TEXT/WEBVTT' is not supported.
[demux] Detected file format: Matroska
[find_files] Loading external files in .
[global] config path: 'sub/' -/-> '/home/pipe/.config/mpv/sub/'
[global] config path: 'sub/' -/-> '/home/pipe/.mpv/sub/'
[global] config path: 'sub/' -/-> '/etc/mpv/sub/'
[global] config path: 'audio/' -/-> '/home/pipe/.config/mpv/audio/'
[global] config path: 'audio/' -/-> '/home/pipe/.mpv/audio/'
[global] config path: 'audio/' -/-> '/etc/mpv/audio/'
[cplayer]  (+) Video --vid=1 (*) (h264)
[cplayer]  (+) Audio --aid=1 (*) (aac)
[cplayer]  (+) Subs  --sid=1 (*) ()
[vo/opengl] Initializing OpenGL backend 'wayland'
[vo/opengl/wayland] failed to connect to a wayland server: check if a wayland compositor is running
[vo/opengl] Initializing OpenGL backend 'x11probe'
[vo/opengl/x11] X11 opening display: :0
[vo/opengl/x11] X11 running at 1366x768 (":0" => local display)
[vo/opengl/x11] Detected wm supports NetWM.
[vo/opengl/x11] Detected wm supports FULLSCREEN state.
[vo/opengl/x11] Detected wm supports ABOVE state.
[vo/opengl/x11] Detected wm supports BELOW state.
[vo/opengl/x11] Display 0 (LVDS1): [0, 0, 1366, 768] @ 60.098447 FPS
[vo/opengl/x11] Current display FPS: 60.098447
[vo/opengl] GLX chose FB config with ID 0x543572c0
[vo/opengl] GLX chose visual with ID 0xb2
[vo/opengl] Creating OpenGL 3.3 context...
[vo/opengl] GL_VERSION='3.3 (Core Profile) Mesa 11.0.6'
[vo/opengl] Detected desktop OpenGL 3.3.
[vo/opengl] GL_VENDOR='Intel Open Source Technology Center'
[vo/opengl] GL_RENDERER='Mesa DRI Intel(R) Ivybridge Mobile '
[vo/opengl] GL_SHADING_LANGUAGE_VERSION='3.30'
[vo/opengl] Loaded extension GLX_SGI_swap_control.
[vo/opengl] Loaded extension GLX_SGI_video_sync.
[vo/opengl] Loaded extension GL_ARB_debug_output.
[vo/opengl] No vdpau support found - probing more things.
[vo/opengl/x11] uninit ...
[vo/opengl] Initializing OpenGL backend 'x11'
[vo/opengl/x11] X11 opening display: :0
[vo/opengl/x11] X11 running at 1366x768 (":0" => local display)
[vo/opengl/x11] Detected wm supports NetWM.
[vo/opengl/x11] Detected wm supports FULLSCREEN state.
[vo/opengl/x11] Detected wm supports ABOVE state.
[vo/opengl/x11] Detected wm supports BELOW state.
[vo/opengl/x11] Display 0 (LVDS1): [0, 0, 1366, 768] @ 60.098447 FPS
[vo/opengl/x11] Current display FPS: 60.098447
[vo/opengl] GLX chose FB config with ID 0x543574f0
[vo/opengl] GLX chose visual with ID 0xb2
[vo/opengl] Creating OpenGL 3.3 context...
[vo/opengl] GL_VERSION='3.3 (Core Profile) Mesa 11.0.6'
[vo/opengl] Detected desktop OpenGL 3.3.
[vo/opengl] GL_VENDOR='Intel Open Source Technology Center'
[vo/opengl] GL_RENDERER='Mesa DRI Intel(R) Ivybridge Mobile '
[vo/opengl] GL_SHADING_LANGUAGE_VERSION='3.30'
[vo/opengl] Loaded extension GLX_SGI_swap_control.
[vo/opengl] Loaded extension GLX_SGI_video_sync.
[vo/opengl] Loaded extension GL_ARB_debug_output.
[vo/opengl] Reported display depth: R=8, G=8, B=8
[vo/opengl] 16 bit texture depth: 16.
[vo/opengl] Testing FBO format 0x805b
[vo/opengl] Create FBO: 16x16 (16x16)
[vo/opengl] Using FBO format 0x805b.
[vo/opengl] No advanced processing required. Enabling dumb mode.
[vo/opengl] Assuming 60.098447 FPS for display sync.
[vd] Container reported FPS: 24.000000
[vd] Codec list:
[vd]     lavc:h264 - H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
[vd] Opening video decoder lavc:h264
[vd] Using software decoding.
[vd] Detected 8 logical cores.
[vd] Requesting 9 threads for decoding.
[vd] Selected video codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10 [lavc:h264]
[vo/opengl/x11] Disabling screensaver.
[ad] Codec list:
[ad]     lavc:aac - AAC (Advanced Audio Coding)
[ad]     lavc:aac_fixed (aac) - AAC (Advanced Audio Coding)
[ad]     lavc:libfdk_aac (aac) - Fraunhofer FDK AAC
[ad] Opening audio decoder lavc:aac
[ad] Requesting 1 threads for decoding.
[ad] Selected audio codec: AAC (Advanced Audio Coding) [lavc:aac]
[sub/ass] Could not open libavcodec subtitle converter
[sub] Could not find subtitle decoder for format ''.
[cplayer] Starting playback...
[af] Audio filter chain:
[af]   [in] 48000Hz stereo 2ch floatp
[af]   [out] 48000Hz stereo 2ch floatp
[af]   [ao] 48000Hz stereo 2ch floatp
[ao] Trying audio driver 'pulse'
[ao/pulse] requested format: 48000 Hz, stereo channels, floatp
[ao/pulse] Library version: 8.0.0
[ao/pulse] Proto: 30
[ao/pulse] Server proto: 4294967295
[ao/pulse] Channel layouts:
[ao/pulse]  - #fl
[ao/pulse]  - #fr
[ao/pulse]  - #fc
[ao/pulse]  - #lfe
[ao/pulse]  - #bl
[ao/pulse]  - #br
[ao/pulse]  - #flc
[ao/pulse]  - #frc
[ao/pulse]  - #bc
[ao/pulse]  - #sl
[ao/pulse]  - #sr
[ao/pulse]  - #tc
[ao/pulse]  - #tfl
[ao/pulse]  - #tfc
[ao/pulse]  - #tfr
[ao/pulse]  - #tbl
[ao/pulse]  - #tbc
[ao/pulse]  - #tbr
[ao/pulse] result: stereo
[ao/pulse] device buffer: 6000 samples.
[ao/pulse] using soft-buffer of 9600 samples.
[cplayer] AO: [pulse] 48000Hz stereo 2ch float
[cplayer] AO: Description: PulseAudio audio output
[af] Adding filter lavrresample 
[af] Audio filter chain:
[af]   [in] 48000Hz stereo 2ch floatp
[af]   [lavrresample] 48000Hz stereo 2ch float
[af]   [out] 48000Hz stereo 2ch float
[af]   [ao] 48000Hz stereo 2ch float
[vd] Decoder format: 1280x720 yuv420p bt.709/limited CL=mpeg2/4/h264
[vd] Using container aspect ratio.
[vf] Video filter chain:
[vf]   [in] 1280x720 yuv420p bt.709/limited CL=mpeg2/4/h264
[vf]   [out] 1280x720 yuv420p bt.709/limited CL=mpeg2/4/h264
[cplayer] VO: [opengl] 1280x720 yuv420p
[cplayer] VO: Description: Extended OpenGL Renderer
[vo/opengl/x11] not waiting for MapNotify
[vo/opengl] Resize: 1280x704
[vo/opengl] Window size: 1280x704
[vo/opengl] Video source: 1280x720 (1:1)
[vo/opengl] Video display: (0, 0) 1280x720 -> (14, 0) 1251x704
[vo/opengl] Video scale: 0.977344/0.977778
[vo/opengl] OSD borders: l=14 t=0 r=15 b=0
[vo/opengl] Video borders: l=14 t=0 r=15 b=0
[vo/opengl] Testing FBO format 0x805b
[vo/opengl] Create FBO: 16x16 (16x16)
[vo/opengl] Using FBO format 0x805b.
[vo/opengl] No advanced processing required. Enabling dumb mode.
[vo/opengl] Texture for plane 0: 1280x720
[vo/opengl] Texture for plane 1: 640x360
[vo/opengl] Texture for plane 2: 640x360
[cplayer] set video colors output-levels=0 
[vo/opengl] recompiling a shader program:
[vo/opengl] [  1] color.r = 1.000000 * vec4(texture(texture0, texcoord0)).r;
[vo/opengl] [  2] color.g = 1.000000 * vec4(texture(texture1, texcoord1)).r;
[vo/opengl] [  3] color.b = 1.000000 * vec4(texture(texture2, texcoord2)).r;
[vo/opengl] [  4] // color conversion
[vo/opengl] [  5] color.rgb = mat3(colormatrix) * color.rgb + colormatrix_c;
[vo/opengl] [  6] color.a = 1.0;
[vo/opengl] [  7] // color management
[osd/libass] fontselect: (sans-serif, 400, 0) -> /usr/share/fonts/dejavu/DejaVuSans.ttf, 0, DejaVuSans
[cplayer] first video frame after restart shown
[cplayer] starting audio playback
[cplayer] playback restart complete
[statusline] AV: 00:00:00 / 00:01:29 (0%) A-V:  0.000
[vo/opengl] Resize: 1366x704
[vo/opengl] Window size: 1366x704
[vo/opengl] Video source: 1280x720 (1:1)
[vo/opengl] Video display: (0, 0) 1280x720 -> (57, 0) 1251x704
[vo/opengl] Video scale: 0.977344/0.977778
[vo/opengl] OSD borders: l=57 t=0 r=58 b=0
[vo/opengl] Video borders: l=57 t=0 r=58 b=0
[statusline] AV: 00:00:01 / 00:01:29 (1%) A-V:  0.000
[cplayer] Run command: quit, flags=9, args=[0]
[cplayer] EOF code: 6  
[ad] Uninit audio decoder.
[af] Removing filter lavrresample 
[vd] Uninit video.
[cplayer] finished playback, success (reason 3)
[cplayer] 
[cplayer] 
[cplayer] Exiting... (Quit)
[ytdl_hook] Exiting...
[osc] Exiting...
[ao/pulse] draining...
[vo/opengl] flushing shader cache
[vo/opengl/x11] Enabling screensaver.
[vo/opengl/x11] uninit ...
```

**Comment by wm4** [comment]
It does support them, but this file is muxed in an unusual/new way. Requires sample.

**Comment by latot** [comment]
file merged with mkvtoolnix 9.2.0

[a.mkv.zip](https://github.com/mpv-player/mpv/files/314077/a.mkv.zip)
